084-templating-host-and-advertisedHost-fields.md

```diff
@@ -0,0 +1,165 @@
+One such example is in the recent [blog post about using the Gateway API](https://strimzi.io/blog/2024/08/16/accessing-kafka-with-gateway-api/):
+
+```yaml
+listeners:
+  - name: obiwan
+    port: 9092
+    tls: true
+    type: cluster-ip
+    configuration:
+      brokerCertChainAndKey:
+        certificate: tls.crt
+        key: tls.key
+        secretName: my-certificate
+      brokers:
+        - advertisedHost: broker-10.strimzi.gateway.api.test
+          advertisedPort: 9092
+          broker: 10
+        - advertisedHost: broker-11.strimzi.gateway.api.test
+          advertisedPort: 9092
+          broker: 11
+        - advertisedHost: broker-12.strimzi.gateway.api.test
+          advertisedPort: 9092
+          broker: 12
+```
+
+## Motivation
+
+Being required to specify these configurations for each broker has some negative side-effects:
+* The `Kafka` custom resource gets unnecessarily complicated, especially for bigger clusters.
+  This also increases the likelihood of typos and other configuration issues.
+* With the introduction of KRaft and node pools, the broker IDs are not always just 0, 1, 2 etc.
+  The brokers might now have different numbers and might include gaps.
+  The controller nodes share the same node ID space, but their hostnames do not need to be configured.
+  All of these easily lead to confusion of how should this configuration look like.
+* When adding new brokers, user needs to make sure the `host` or `advertisedHost` fields will be added for the new brokers before scaling the cluster up.
+
+But in most cases, the (advertised) hostnames often follow the same pattern.
+In the example above, you can see that the Ingress based listener uses the hostname pattern `my-cluster-kafka-<NODE-ID>.myingress.com`.
+And the customized listener uses the advertised hostname pattern `broker-<NODE-ID>.strimzi.gateway.api.test`.
+
+While there might be users using more random hostnames, for many Strimzi users, having an easy way to template the hostnames as suggested in this proposal would provide a huge simplification:
+* Users will be able to scale-up without changing the configuration first
+* The `Kafka` CR will be much smaller, simpler and easier to read
+* The likelihood of typos and misconfigurations will be smaller
+
+## Proposal
+
+This proposal suggests adding two new fields to the `configuration` section of the listener:
+* `advertisedHostTemplate`
+* `hostTemplate`
+
+These fields will be strings that will be used to configure templates that will be used to generate the hostnames.
+The templates will support several different _variables_:
+* `{nodeId}` will be replaced with the ID of the Kafka node to which the template is applied
+* `{nodeName}` will be replaced with the name of the Kafka node (name of the Kubernetes Pod used for this node) to which the template is applied
```

Comment:
Generally a node can be thought of as having several 'names'. There's the pod name, but also host names (with and without qualification). It not really ambiguous here, because we're _configuring_ the host names. But if someone looked at the CR with the assumption that the `nodeName` was a host name then there's potential for confusion.

Another point to bear in mind, if the proposal is accepted then maybe we end up with other templates in CRs. Ideally we'd use a common set of variable names across all those places. In those other places `nodeName` might be more ambiguous.

For these reasons perhaps `nodePodName` would be better?

Reply:
This makes sense. I updated it to `nodePodName`.